lib/receptor_controller/client/response_worker.rb

```diff
@@ -95,7 +95,7 @@ def process_message(message)
       response = JSON.parse(message.payload)
 
       if (message_id = response['in_response_to'])
-        logger.debug("Receptor response: Received message_id: #{message_id}")
+        logger.debug("Receptor response: Received message_id: #{message_id}; type: #{response['message_type']}")
```

Comment:
@slemrmartin Can you also log the queue_persist_ref so we can track it from the logs. Also there needs to be cleanup when the pod is terminated. if the manageiq-messaging gem doesn't handle SIG_TERM we might have to directly use the ruby-kafka gem and handle the SIG_TERM like its mentioned here
https://github.com/zendesk/ruby-kafka#shutting-down-a-consumer

Reply:
The messaging gem traps SIG_TERM.